addons/web/controllers/json.py

```diff
@@ -0,0 +1,254 @@
+            else:
+                default_domain = expression.AND(map(
+                    context_eval,
+                    self.__filter_from_context(context, action, model),
+                ))
+            if default_domain and default_domain != expression.TRUE_DOMAIN:
+                kwargs["domain"] = repr(default_domain)
+            user_domain = default_domain
+        try:
+            limit = int(kwargs.get("limit", 0)) or action.limit
+            offset = int(kwargs.get("offset", 0))
+        except ValueError as exc:
+            raise BadRequest(exc.args[0])
+        if "offset" not in kwargs:
+            kwargs["offset"] = offset
+        if "limit" not in kwargs:
+            kwargs["limit"] = limit
+
+        # Add date domain for some view types
+        date_domain = []
+        if view_type in ('calendar', 'gantt', 'cohort'):
+            try:
+                start_date, end_date = self.__get_start_date_end_date(kwargs)
+            except ValueError as exc:
+                raise BadRequest(exc)
+            if start_date and end_date:
+                date_field = view_tree.attrib.get("date_start")
+                assert date_field, "Could not find the date field in the view"
+                date_domain = [(date_field, '>=', start_date), (date_field, '<', end_date)]
+            else:
+                # by default, select the current month
+                month = date.today() + relativedelta(day=1)
+                kwargs.update({
+                    'start_date': month.isoformat(),
+                    'end_date': (month + relativedelta(months=1)).isoformat(),
+                })
+
+        # Add explicitly activity fields for an activity view
+        activity_domain = []
+        if view_type == 'activity':
+            activity_domain = [('activity_ids', '!=', False)]
+            # add activity fields
+            for field_name, field in model._fields.items():
+                if field_name.startswith("activity_") and field_name not in spec and field.is_accessible(env):
+                    spec[field_name] = {}
+
+        # Group by
+        groupby = kwargs.get('groupby')
+        fields = kwargs.get('fields')
+        if groupby:
+            groupby = groupby.split(",")
+        if fields:
+            fields = fields.split(',')
+        if groupby is None and view_tree.attrib.get('default_group_by'):
+            # in case the kanban view (or other) defines a default grouping
+            # add that field to the spec
+            default_group_by_field = view_tree.attrib.get('default_group_by')
+            spec.setdefault(default_group_by_field, {})
+        if not groupby and view_type in ('pivot', 'graph'):
+            # extract groupby from the view if we don't have any
+            field_by_type = defaultdict(list)
+            for element in view_tree.findall(r"./field"):
+                field_name = element.attrib.get("name")
+                if int(element.attrib.get("invisible", 0)):
```

Comment:
The attribute `invisible` may contain Python expressions (for list views, for instance).  In such a case, this line will crash.
```suggestion
                if element.attrib.get("invisible", "") in ("1", "True"):
```